supervisor/host/network.py

```diff
@@ -44,3 +78,237 @@ async def update(self):
         except DBusNotConnectedError as err:
             _LOGGER.error("No hostname D-Bus connection available")
             raise HostNotSupportedError() from err
+
+    async def apply_changes(self, interface: Interface) -> None:
+        """Apply Interface changes to host."""
+        inet = self.sys_dbus.network.interfaces.get(interface.name)
+
+        # Update exist configuration
+        if inet and inet.settings and interface.enabled:
+            settings = interface_update_payload(
+                interface,
+                name=inet.settings.connection.id,
+                uuid=inet.settings.connection.uuid,
+            )
+
+            try:
+                await inet.settings.update(settings)
+            except DBusError as err:
+                _LOGGER.error("Can't update config on %s", interface.name)
+                raise HostNetworkError() from err
+
+        # Create new configuration and activate interface
+        elif inet and interface.enabled:
+            settings = interface_update_payload(interface)
+
+            try:
+                await self.sys_dbus.network.add_and_activate_connection(
+                    settings, inet.object_path
+                )
+            except DBusError as err:
+                _LOGGER.error("Can't create config and activate %s", interface.name)
+                raise HostNetworkError() from err
+
+        # Remove config from interface
+        elif inet and not interface.enabled:
+            try:
+                await inet.settings.delete()
+            except DBusError as err:
+                _LOGGER.error("Can't remove %s", interface.name)
+                raise HostNetworkError() from err
+
+        # Create new interface (like vlan)
+        elif not inet:
+            settings = interface_update_payload(interface)
+
+            try:
+                await self.sys_dbus.network.settings.add_connection(settings)
+            except DBusError as err:
+                _LOGGER.error("Can't create new interface")
+                raise HostNetworkError() from err
+
+        await self.update()
+
+    async def scan_wifi(self, interface: Interface) -> List[AccessPoint]:
+        """Scan on Interface for AccessPoint."""
+        inet = self.sys_dbus.network.interfaces.get(interface.name)
+
+        if inet.type != DeviceType.WIRELESS:
+            _LOGGER.error("Can only scan with wireless card - %s", interface.name)
+            raise HostNotSupportedError()
+
+        await inet.wireless.request_scan()
+        await asyncio.sleep(5)
```

Comment:
Should this be configurable?
In a chatty env, 5s might not be enough

Reply:
Future, we listen to the dbus signal for update properties, but the PR is enough big. He scan also constantly in background. We have 100 warranty to see almost everything after 1sec. 5 should be fine until we add the signal listener